dot/parachain/prospective-parachains/prospective_parachains_test.go

```diff
@@ -910,3 +914,114 @@ func TestAnswerProspectiveValidationDataRequest(t *testing.T) {
 	// Close the channels
 	close(subsystemToOverseer)
 }
+
+func markCandidatedBacked(
+	t *testing.T,
+	overseerToSubsystem chan any,
+	candidate parachaintypes.CommittedCandidateReceipt,
+) {
+	hash, err := candidate.Hash()
+
+	assert.NoError(t, err)
+
+	msg := messages.CandidateBacked{
+		ParaID:        candidate.Descriptor.ParaID,
+		CandidateHash: parachaintypes.CandidateHash{Value: hash},
+	}
+
+	overseerToSubsystem <- msg
+}
+
+func TestHandleBacked(
+	t *testing.T,
+) {
+	candidateRelayParent := common.Hash{0x01}
+	paraId := parachaintypes.ParaID(1)
+	parentHead := parachaintypes.HeadData{
+		Data: bytes.Repeat([]byte{0x01}, 32),
+	}
+	headData := parachaintypes.HeadData{
+		Data: bytes.Repeat([]byte{0x02}, 32),
+	}
+	validationCodeHash := parachaintypes.ValidationCodeHash{0x01}
+	candidateRelayParentNumber := uint32(0)
+
+	candidate := makeCandidate(
+		candidateRelayParent,
+		candidateRelayParentNumber,
+		paraId,
+		parentHead,
+		headData,
+		validationCodeHash,
+	)
+
+	pvd := dummyPVD(parentHead, candidateRelayParentNumber)
+
+	subsystemToOverseer := make(chan any)
+	overseerToSubsystem := make(chan any)
+
+	prospectiveParachains := NewProspectiveParachains(subsystemToOverseer)
+
+	relayParent := relayChainBlockInfo{
+		Hash:        candidateRelayParent,
+		Number:      0,
+		StorageRoot: common.Hash{0x00},
+	}
+
+	baseConstraints := &parachaintypes.Constraints{
+		RequiredParent:       parachaintypes.HeadData{Data: bytes.Repeat([]byte{0x01}, 32)},
+		MinRelayParentNumber: 0,
+		ValidationCodeHash:   validationCodeHash,
+		MaxPoVSize:           1000000,
+	}
+
+	scope, err := newScopeWithAncestors(relayParent, baseConstraints, nil, 10, nil)
+	assert.NoError(t, err)
+
+	prospectiveParachains.View.perRelayParent[candidateRelayParent] = &relayParentData{
+		fragmentChains: map[parachaintypes.ParaID]*fragmentChain{
+			paraId: newFragmentChain(scope, newCandidateStorage()),
+		},
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func(wg *sync.WaitGroup) {
```

Comment:
nit: I don't think we need to pass in `*sync.WaitGroup` to this anon goroutine given we are in the test scope.

Reply:
oh, but that is just a common practice, I would prefer to pass args to the goroutine than using an outer one